app/src/main/java/com/willowtree/vocable/presets/AddToCategoryPickerFragment.kt

```diff
@@ -0,0 +1,162 @@
+import androidx.viewpager2.widget.ViewPager2
+import com.willowtree.vocable.*
+import com.willowtree.vocable.databinding.FragmentAddToCategoryPickerBinding
+import com.willowtree.vocable.presets.adapter.AddToCategoryPickerViewModel
+import com.willowtree.vocable.room.Category
+import com.willowtree.vocable.settings.SettingsActivity
+import com.willowtree.vocable.utils.VocableFragmentStateAdapter
+import kotlin.math.min
+
+class AddToCategoryPickerFragment : BaseFragment<FragmentAddToCategoryPickerBinding>() {
+
+    private val args: AddToCategoryPickerFragmentArgs by navArgs()
+
+    override val bindingInflater: BindingInflater<FragmentAddToCategoryPickerBinding> =
+        FragmentAddToCategoryPickerBinding::inflate
+    private lateinit var addToCategoryPickerViewModel: AddToCategoryPickerViewModel
+
+    private var maxCategories = 1
+    private lateinit var categoriesAdapter: CategoriesPagerAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val phraseText = args.phraseText
+
+        binding.backButton.action = {
+            findNavController().popBackStack()
+        }
+
+        val numColumns = resources.getInteger(R.integer.custom_category_columns)
+        val numRows = resources.getInteger(R.integer.custom_category_rows)
+        maxCategories = numColumns * numRows
+
+        categoriesAdapter = CategoriesPagerAdapter(childFragmentManager)
+
+        binding.categoryPagerForwardButton.action = {
+            when (val currentPosition = binding.categoryHolder.currentItem) {
+                categoriesAdapter.itemCount - 1 -> {
+                    binding.categoryHolder.setCurrentItem(0, true)
+                }
+                else -> {
+                    binding.categoryHolder.setCurrentItem(currentPosition + 1, true)
+                }
+            }
+        }
+
+        binding.categoryPagerBackButton.action = {
+            when (val currentPosition = binding.categoryHolder.currentItem) {
+                0 -> {
+                    binding.categoryHolder.setCurrentItem(
+                        categoriesAdapter.itemCount - 1,
+                        true
+                    )
+                }
+                else -> {
+                    binding.categoryHolder.setCurrentItem(currentPosition - 1, true)
+                }
+            }
+        }
+
+        binding.categoryHolder.registerOnPageChangeCallback(object :
+            ViewPager2.OnPageChangeCallback() {
+            override fun onPageSelected(position: Int) {
+                val pageNum = position % categoriesAdapter.numPages + 1
+                binding.categoryPageNumber.text = getString(R.string.phrases_page_number, pageNum, categoriesAdapter.numPages)
+
+                val activity = requireActivity()
+                if (activity is MainActivity) {
+                    activity.resetAllViews()
+                }
+            }
+        })
```

Comment:
Since you only care about `onPageSelected(position: Int)`, you could probably create an extension function to make this more readable and re-usable elsewhere. Something like...

```
fun ViewPager2.onPageSelected(onPageSelected: (Int) -> Unit) {
    registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
        override fun onPageSelected(position: Int) {
            super.onPageSelected(position)
            onPageSelected(position)
        }
    })
}
...

// So then you could just call something like
binding.categorHolder.onPageSelected { position ->
    // Do your thangs here
}
```

Or since yall like using lambdas and what not. You could even do some sort of extension val instead.
```
binding.categoryHolder.onPageSelected = { position ->
    // Do thangs
}
```